src/github_repo_request_local.py

```diff
@@ -0,0 +1,125 @@
+import argparse
+import subprocess
+from pathlib import Path
+import pandas as pd
+from loguru import logger
+import shutil
+from utils.git_utils import get_working_directory_or_git_root
+
+"""
+This script automates the process of cloning GitHub repositories listed in a
+CSV file, counts the number of test files in each repository, and saves the
+count back to the CSV. It's designed to handle interruptions by saving progress
+incrementally and resuming where it left off. Users can specify excluded file
+extensions and choose a custom clone directory.
+"""
+
+
+def parse_args():
+    """Parse command line arguments for excluded extensions and clone
+    directory."""
+    parser = argparse.ArgumentParser(
+        description="Clone GitHub repositories and count " "test files."
+    )
+    parser.add_argument(
+        "--exclude",
+        nargs="+",
+        default=[".txt", ".md", ".h", ".xml", ".html", ".json", ".png", ".jpg"],
+        help="File extensions to exclude. Pass each extension as a"
+        " separate argument prefixed by --exclude.",
+    )
+    parser.add_argument(
+        "--clone-dir",
+        type=str,
+        default=str(Path.home() / "data" / "cloned_repos"),
+        help="Directory to clone repositories into. Defaults to "
+        "~/data/cloned_repos.",
+    )
+    parser.add_argument(
+        "--keep-clones",
+        action="store_true",
+        help="Keep cloned repositories after processing. If not specified, "
+        "cloned repositories ill be deleted.",
+    )
+    return parser.parse_args()
+
+
+def list_test_files(directory, excluded_extensions):
+    """List test files in the directory, excluding specified extensions."""
+    logger.info(f"Processing the directory {directory}")
+    test_files = []
+
+    for item in directory.rglob("*"):
+        # Check if the item is a file and either the file or its parent
+        # directory contains 'test'
+        if item.is_file() and (
+            "test" in item.name.lower() or "test" in str(item.parent).lower()
+        ):
+            # Exclude files with certain extensions
+            if item.suffix not in excluded_extensions:
+                test_files.append(str(item))
+    return test_files
+
+
+args = parse_args()
+# Log the excluded file extensions
+logger.info(f"Excluded file extensions: {', '.join(args.exclude)}")
+
+# Use git_codebase_root to define paths relative to the repository root
```

Comment:
This comment is slightly out of date now you're calling the new function, just change the name of the function in the comment to fix.

Reply:
This is fixed.